### PR TITLE
Add coverage report to CI

### PR DESCRIPTION
**Pull request by PurpleMyst**

### CODE REVIEW

File: convert-coverage.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+"""Convert a cov.xml report so that it is usable locally"""
```
Comment (Akuli) [inline convert-coverage.py:1]: Which tools can read cov.xml after running this script?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -29,4 +29,8 @@ jobs:
       - run: |
           pip install poetry && poetry install
       - run: |
-          poetry run pytest -v tests.py
+          poetry run pytest --cov=sansio_lsp_client --cov-report=xml:cov.xml -v tests.py
+      - uses: actions/upload-artifact@v2
+        with:
+          name: coverage-report
+          path: cov.xml
```
Comment (Akuli) [inline .github/workflows/test.yml:36]: Where can the resulting xml files be viewed? What tools can be used to look at them?

File: convert-coverage.py (hunk 1 above)
Comment (PurpleMyst) [reply convert-coverage.py:1]: [This](https://github.com/ryanluker/vscode-coverage-gutters) is the one I'm utilizing, but I believe it to be a standard format that many tools understand.

File: .github/workflows/test.yml (hunk 2 above)
Comment (PurpleMyst) [reply .github/workflows/test.yml:36]: See previous comment

**Review by Akuli — APPROVED** [review]
I think I will just run the tests with coverage on my system and ask coverage to output HTML, but the xml uploading alternative seems like it is useful for you.

**Comment by PurpleMyst** [comment]
WDYM by output HTML? I'd no idea that was an option :o

**Comment by Akuli** [comment]
I usually run Porcupine tests with `pytest --cov=porcupine`, and then I run `coverage html`. I then open `htmlcov/index.html` in the browser and look at the coverages.

**Comment by PurpleMyst** [comment]
Ah, that's cool

**Pull request merged**